lib/my_app.rb

```diff
@@ -0,0 +1,151 @@
+ENV['RACK_ENV'] ||= "development"
+
+require_relative '../data_mapper_setup.rb'
+require 'sinatra/base'
+require 'sinatra/flash'
+
+class MakersBnb < Sinatra::Base
+  use Rack::MethodOverride
+  register Sinatra::Flash
+
+  enable :sessions
+  set :session_secret, 'super secret'
+
+  helpers do
+    def current_user
+      @current_user ||= User.get(session[:user_id]) 
+    end
+  end
+
+  get '/' do
+    @spaces = Space.all
+    erb :index
+  end
+
+  post '/' do
+    a_space = Space.create(params)  
+    current_user.space << a_space
+    current_user.save
+    redirect to('/new')
+  end
+
+  get '/new' do 
+    erb :newPost
+  end
+
+  get '/users/new' do
+    @user = User.new
+    erb :new_user
+  end
+
+  post '/users' do
+    @user = User.create(email: params[:email],
+                        password: params[:password],
+                        password_confirmation: params[:password_confirmation],
+                        user_name: params[:user_name])
+    if @user.save
+      session[:user_id] = @user.id
+      redirect to '/'
+    else
+      flash.now[:errors] = @user.errors.full_messages
+      erb :new_user
+    end
+  end
+
+  get '/sessions/new' do
+    erb :'sessions/new'
+  end
+
+  get '/booked/:id' do 
+    space = Space.last(id: params[:id])
+    if current_user != nil
+      owner = space.user.last.user_name
+      user = current_user.user_name
+      @booking_object = Booking.create(name: user, owner: owner, room: space.name, confirmation: params[:false])
+      session[:booking_ex] = @booking_object.id
+      @booking = Booking.last(id: session[:booking_ex])
+    end
+      erb :confirmation
+  end
```

Comment:
Definitely too long as a controller action, and a few code smells to chase down here.

1. `if x != nil` is essentially the same as `if x`
2. Lines 64 - 66 seem to be doing some black magic to make the Booking work. This doesn't look like a healthy way to do it, and needs some more design.

What happens if I just visit this route as a non-logged-in user? A few things to look at here.